BTCPayServer/Plugins/PluginService.cs

```
@@ -49,32 +55,94 @@ public Version GetVersionOfPendingInstall(string plugin)
-                p.Documentation = v.Documentation;
-                var github = v.BuildInfo.GetGithubRepository();
-                if (github != null)
-                {
-                    p.Source = github.GetSourceUrl(v.BuildInfo.gitCommit, v.BuildInfo.pluginDir);
-                    p.Author = github.Owner;
-                    p.AuthorLink = $"https://github.com/{github.Owner}";
-                }
-                p.SystemPlugin = false;
-                return p;
-            }).ToArray();
+                plugins.AddRange(
+                    updates.Select(MapToAvailablePlugin)
+                        .Where(p => p is not null)
+                        .Select(p => p!)
+                );
+            }
+
+            return plugins.ToArray();
+        }
+
+#nullable enable
+        private AvailablePlugin? MapToAvailablePlugin(PublishedVersion publishedVersion)
+        {
+            if (publishedVersion.ManifestInfo is null)
+            {
+                _logger.LogWarning("Skipping published plugin with missing ManifestInfo (BuildId: {BuildId}, ProjectSlug: {ProjectSlug})", publishedVersion.BuildId, publishedVersion.ProjectSlug);
+                return null;
+            }
+
+            AvailablePlugin? availablePlugin;
+
+            try
+            {
+                availablePlugin = publishedVersion.ManifestInfo.ToObject<AvailablePlugin>();
+            }
+            catch (JsonException ex)
+            {
+                _logger.LogWarning(ex, "Failed to parse ManifestInfo (BuildId: {BuildId}, ProjectSlug: {ProjectSlug})", publishedVersion.BuildId, publishedVersion.ProjectSlug);
+                return null;
+            }
+
+            if (availablePlugin is null)
+            {
+                _logger.LogWarning("Failed to deserialize ManifestInfo to AvailablePlugin (BuildId: {BuildId}, ProjectSlug: {ProjectSlug})", publishedVersion.BuildId, publishedVersion.ProjectSlug);
```

don't log, let it crash.